src/helix/client/client_ext.rs

```diff
@@ -563,6 +563,40 @@ impl<'client, C: crate::HttpClient + Sync + 'client> HelixClient<'client, C> {
         make_stream(req, token, self, std::collections::VecDeque::from)
     }
 
+    /// Gets a list of channels that the specified user has moderator privileges in. [Get Moderated Channels](helix::moderation::GetModeratedChannelsRequest)
+    ///
+    /// `user_id` must match the user ID in the User-Access token (`token`).
+    ///
+    /// # Examples
+    ///
+    /// ```rust, no_run
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    /// # let client: helix::HelixClient<'static, twitch_api::client::DummyHttpClient> = helix::HelixClient::default();
+    /// # let token = twitch_oauth2::AccessToken::new("validtoken".to_string());
+    /// # let token = twitch_oauth2::UserToken::from_existing(&client, token, None, None).await?;
+    /// use twitch_api::helix;
+    /// use futures::TryStreamExt;
+    ///
+    /// let requests: Vec<helix::moderation::ModeratedChannel> = client.get_moderated_channels("1234", &token).try_collect().await?;
+    /// # Ok(()) }
+    /// ```
+    pub fn get_moderated_channels<'b: 'client, T>(
+        &'client self,
+        user_id: impl types::IntoCow<'b, types::UserIdRef> + 'b,
```

Comment:
The token already contains the user id. I think we should take the user id from it

Reply:
I `.expect` the `user_id` currently. Otherwise, the method would return an `Option<impl Stream...>`. Though, I know that for my tokens, I don't store the user-id.

Reply:
hmm, maybe we should reconsider this, take for example `get_followed_streams` it also uses the user-id on the token. Is there a reason you're not attaching the user-id?

Reply:
> Is there a reason you're not attaching the user-id?

Nothing in particular. I never had to use it from there. I only needed an access token, refresh token, and a client-id.

> hmm, maybe we should reconsider this, take for example `get_followed_streams` it also uses the user-id on the token.

Maybe. Having a method that takes a user-id is the most generic. If the user knows their token has a user-id, they can use `token.user_id().unwrap()`.

Reply:
Yes, I think I agree, we can showcase that in the documentation also. Lets switch to that for the future and existing methods